runtime/compiler/aarch64/codegen/J9TreeEvaluator.cpp

```diff
@@ -322,8 +322,174 @@ J9::ARM64::TreeEvaluator::DIVCHKEvaluator(TR::Node *node, TR::CodeGenerator *cg)
       snippet->gcMap().setGCRegisterMask(0xffffffff);
       }
 
-   cg->evaluate(node->getFirstChild());
+TR::Register *
+J9::ARM64::TreeEvaluator::ArrayCHKEvaluator(TR::Node *node, TR::CodeGenerator *cg)
+   {
+   TR::Register *obj1Reg = cg->evaluate(node->getFirstChild());
+   TR::Register *obj2Reg = cg->evaluate(node->getSecondChild());
+   TR::Register *tmp1Reg = cg->allocateRegister();
+   TR::Register *tmp2Reg = cg->allocateRegister();
+   TR::LabelSymbol *doneLabel = generateLabelSymbol(cg);
+   TR::RegisterDependencyConditions *deps = new (cg->trHeapMemory()) TR::RegisterDependencyConditions(2, 2, cg->trMemory());
```

Comment:
The size of the allocated `RegisterDependencyConditions` is wrong.  You call `TR::addDependency()` on 4 registers below.